Missing `API` type in component.d.ts

-   Laravel Mix Version: 6.0.9
-   Node Version: v15.12.0
-   NPM Version: 7.6.3
-   OS: Windows 10 (WSL2 w/ Ubuntu 20.04)

### Description: 

When compiling a Typescript component into js (to be published on npm) `tsc` complains about a missing type in the component.d.ts

```bash
node_modules/laravel-mix/types/component.d.ts:76:11 - error TS2304: Cannot find name 'API'.

76     (mix: API, config: webpack.Configuration, ...args: any[]): void;
             ~~~


Found 1 error.
```
I've tried to see if I could import API from the index file in the component.d.ts file, but didn't manage to make it work so far.

### Steps To Reproduce:

Add typescript as a dependency, add a tsconfig file and run the `tsc` command.

This branch can be used to replicate the problem: https://github.com/Elhebert/laravel-mix-sri/tree/typescript-rewrite

--- a/types/component.d.ts
+++ b/types/component.d.ts
@@ -1,7 +1,8 @@
 // Type definitions for laravel-mix 6.0
 
 import * as webpack from 'webpack';
-import { TransformOptions as BabelConfig } from 'babel-core';
+import { TransformOptions as BabelConfig } from '@babel/core';
+import api from './index';
 
 export type DependencyObject = {
     /** The name of the package */
@@ -73,7 +74,7 @@ export interface ClassComponent {
 }
 
 export interface FunctionalComponent {
-    (mix: Api, config: webpack.Configuration, ...args: any[]): void;
+    (mix: typeof api, config: webpack.Configuration, ...args: any[]): void;
 }
 
 export type Component = ClassComponent | FunctionalComponent;

--- a/types/config.d.ts
+++ b/types/config.d.ts
@@ -8,7 +8,7 @@ import { Options as GifsicleConfig } from 'imagemin-gifsicle';
 import { Options as MozjpegConfig } from 'imagemin-mozjpeg';
 import { Options as OptipngConfig } from 'imagemin-optipng';
 import { Options as SvgoConfig } from 'imagemin-svgo';
-import { TransformOptions as BabelConfig } from 'babel-core';
+import { TransformOptions as BabelConfig } from '@babel/core';
 // import { TerserPluginOptions } from 'terser-webpack-plugin';
 import { TerserPluginOptions } from './terser';
 import { AcceptedPlugin } from 'postcss';

